openrtb_ext/alternatebiddercodes.go

```diff
@@ -0,0 +1,53 @@
+package openrtb_ext
+
+import "fmt"
+
+// ExtAlternateBidderCodes defines list of alternate bidder codes allowed by adatpers. This overrides host level configs.
+type ExtAlternateBidderCodes struct {
+	Enabled bool                                      `mapstructure:"enabled" json:"enabled"`
+	Bidders map[string]ExtAdapterAlternateBidderCodes `mapstructure:"bidders" json:"bidders"`
+}
+
+type ExtAdapterAlternateBidderCodes struct {
+	Enabled            bool     `mapstructure:"enabled" json:"enabled"`
+	AllowedBidderCodes []string `mapstructure:"allowedbiddercodes" json:"allowedbiddercodes"`
+}
+
+func (bidderCodes *ExtAlternateBidderCodes) IsValidBidderCode(bidder, alternateBidder string) (bool, error) {
+	const ErrAlternateBidderNotDefined = "alternateBidderCodes not defined for adapter %q, rejecting bids for %q"
+	const ErrAlternateBidderDisabled = "alternateBidderCodes disabled for %q, rejecting bids for %q"
```

Comment:
The naming convention `errSomeName` is usually for when the value is an error type. This is an error message and the separation of the fields from the Errorf calls is a bit hard to follow IMHO. Consider extracting each to their own simple builder method, like:

```
func alternateBidderNotDefinedError(bidder, alternateBidder string) error {
  return fmt.Errorf("alternateBidderCodes not defined for adapter %q, rejecting bids for %q", bidder, alternateBidder)
}
```

Reply:
done